solarforecastarbiter/metrics/calculator.py

```diff
@@ -503,15 +487,17 @@ def _transform_prob_forecast_value_and_prob(proc_fx_obs,
         Forecast physical values and forecast probabilities.
     """
     fx_fx_prob = []
-    df = getattr(proc_fx_obs, attr)
-    assert isinstance(df, pd.DataFrame)
-    for col in df.columns:
+    data = getattr(proc_fx_obs, attr)
+    if isinstance(data, pd.Series):
+        data = data.to_frame()
+
+    for col in data.columns:
         if proc_fx_obs.original.forecast.axis == 'x':
-            fx_prob = df[col]
+            fx_prob = data[col]
             fx = pd.Series([float(col)]*fx_prob.size, index=fx_prob.index,
                            name=col)
         else:  # 'y'
-            fx = df[col]
+            fx = data[col]
             fx_prob = pd.Series([float(col)]*fx.size, index=fx.index,
```

Comment:
I ran into an exception here

```
Exception has occurred: ValueError
could not convert string to float: 'value'
```

Reply:
When are you running into this error? I just reran the metrics tests and didn't get an error. I haven't tested any reports though

Reply:
Just want to make sure if this needs to be addressed now if so how to I reproduce it?

Reply:
I was using this as a test forecast:

https://dashboard.solarforecastarbiter.org/forecasts/cdf/single/a1fc6490-080a-11ea-a9db-0a580a8003a5

`data` is a DataFrame with a single column `value`, so the string conversion fails.

I think the issue with the tests is that `test_calculate_probabilistic_metrics` is that it doesn't test series input like you'd get with the `ProbabilisticForecastConstantValues`. I tried adding this parameterization but ran into a lot more errors than I wanted to deal with.

```
    ('y',
     pd.Series(np.random.randn(10)*np.sqrt(20)+20),
     pd.Series(np.random.randn(10)*np.sqrt(20)+20),
     pd.Series(np.random.randn(10)*np.sqrt(20)+20, dtype=float)),
```

Reply:
I just ran through the `test_calculate_probabilistic_metrics` tests in the portion where it does the constant value tests:

https://github.com/awig/solarforecastarbiter-core/blob/952afd7bc4cbcd04b841165d2f18c6a9165c0f25/solarforecastarbiter/metrics/tests/test_calculator.py#L761

When I grab the column from the dataframe it gets a `pandas.Series` and verified that the `ProcessedForecastObservation.forecast_values` is a `pandas.Series` as well. 

```
(Pdb) type(cv_pair.forecast_values)
<class 'pandas.core.series.Series'>
```

The errors you were getting were most likely because of the setup particular to the test.

It should also be tested in `test_calculate_probabilistic_metrics_with_constant_value_simple`. Where the test takes a dataframe but then they get converted to series before passing to `calculate_probabilistic_metrics`. 

But just so I know how it exactly you are doing it are you trying to add to the tests or running a report via the CLI? Is it a conversion issue in `calculator._transform_prob_forecast_value_and_prob`?  That's the only place i can think that is referencing the columns but that is through an iterator not by column name in particular.

Reply:
I was trying to create a report via the CLI and ran into the issue. From that, it looked to me like the tests might have been missing an important condition, but yes, I ran into test setup issues when trying to add the condition myself.